### ISSUE TITLE
Error response from daemon: NetworkDriver.CreateNetwork: Non-Calico IPAM driver is used

### ISSUE THREAD

**Issue by tomdee** [projectcalico/libnetwork-plugin#77]
This error is hit when networks are created without the --subnet argument.

```2016/11/05 17:31:29 RequestPool JSON={"AddressSpace":"CalicoGlobalAddressSpace","Pool":"","SubPool":"","Options":{},"V6":false}
2016/11/05 17:31:29 RequestPool response JSON={"PoolID":"CalicoPoolIPv4","Pool":"0.0.0.0/0","Data":{"com.docker.network.gateway":"0.0.0.0/0"}}
2016-11-05 17:31:29.223445 I | Entering go-plugins-helpers createnetwork
2016/11/05 17:31:29 CreateNetwork JSON={"NetworkID":"d169372db9a4dc0e3937c95aa9adaa34acd001b395a7609527c7b93cac55a464","Options":{"com.docker.network.enable_ipv6":false,"com.docker.network.generic":{}},"IPv4Data":[{"AddressSpace":"","Pool":"0.0.0.0/0","Gateway":"0.0.0.0/0","AuxAddresses":null}],"IPv6Data":[]}
2016/11/05 17:31:29 Non-Calico IPAM driver is used
```

**Comment by tomdee** [comment]
isn't a bug on Docker 1.12.3 but it does show up on Docker 1.11.2

**Comment by tomdee** [comment]
Logs from 1.12.3
```
016/11/05 18:26:43 GetCapabilities response JSON={"Scope":"global"}
2016/11/05 18:26:43 GetCapabilities response JSON={"RequiresMACAddress":false}
2016/11/05 18:26:43 GetDefaultAddressSpace response JSON={"LocalDefaultAddressSpace":"CalicoLocalAddressSpace","GlobalDefaultAddressSpace":"CalicoGlobalAddressSpace"}
2016/11/05 18:26:43 RequestPool JSON={"AddressSpace":"CalicoGlobalAddressSpace","Pool":"","SubPool":"","Options":{},"V6":false}
2016/11/05 18:26:43 RequestPool response JSON={"PoolID":"CalicoPoolIPv4","Pool":"0.0.0.0/0","Data":{"com.docker.network.gateway":"0.0.0.0/0"}}
2016-11-05 18:26:43.719963 I | Entering go-plugins-helpers createnetwork
2016/11/05 18:26:43 CreateNetwork JSON={"NetworkID":"1bd5a7e3d4a17bfa119d162375a2743b79aa58bf4c7bc5ca18831b6516356ec5","Options":{"com.docker.network.enable_ipv6":false,"com.docker.network.generic":{}},"IPv4Data":[{"AddressSpace":"CalicoGlobalAddressSpace","Pool":"0.0.0.0/0","Gateway":"0.0.0.0/0","AuxAddresses":null}],"IPv6Data":[]}
```

Crucial difference is in the NetworkCreate which includes the `CalicoGlobalAddressSpace`

**Comment by tomdee** [comment]
Fixed by #78
